docs/adr/adr-11-blocksync-overhaul.md

```diff
@@ -0,0 +1,372 @@
+___[CAR format](https://ipld.io/specs/transport/car/carv2/)___ and ___[Dagstore](https://github.com/filecoin-project/dagstore)___
+for ___extended block storage___ and ___custom p2p Req/Resp protocol for block data syncing___(whole block and data by
+namespace id) in the happy path. The p2p portion of the document will come in the subsequent Part II document.
+
+#### Key Design Decisions
+
+- __FNs/BNs store EDSes serialized as [CAR files](https://ipld.io/specs/transport/car).__ CAR format provides an
+efficient way to store Merkle DAG data, like EDS with NMT. It packs such DAG data into a single blob which can be read
+sequentially in one read and transferred over the wire. Additionally, [CARv2](https://ipld.io/specs/transport/car/carv2/)
+introduces pluggable indexes over the blob allowing efficient random access to shares and NMT Proofs in one read
+(if the index is cached in memory).
+
+- __FNs/BNs manage a top-level index for _hash_ to _CAR block file_ mapping.__ Current DASing for LNs requires FNs/BNs
+to serve simple hash to data requests. The top-level index maps any Share/NMT Proof hash to any block CARv1 file so
+that FNs/BNs can quickly serve DA requests.
+
+- __FNs/BNs address EDSes by `DataRoot.Hash`.__ The only alternative is by height; however, it does not allow block
+data deduplication in case EDSes are equal and couples the Data layer/pkg with the Header layer/pkg.
+
+- __FNs/BNs run a single instance of [`DAGStore`](https://github.com/filecoin-project/dagstore) to manage CAR block
+files.__ DAGStore provides the top-level indexing and CARv2-based indexing per each CAR file. In essence, it's an
+engine for managing any CAR files with indexing, convenient abstractions, tools, recovery mechanisms, etc.
+  - __EDSes as _CARv1_ files over _CARv2_.__ CARv2 encodes indexes into the file, while DAGStore maintains CARv2-based
+indexing. Usage of CARv1 keeps only one copy of the index, stores/transfers less metadata per EDS, and simplifies
+reading EDS from a file.
+
+- __LNs DASing remains untouched__. The networking protocol and storage for LNs remain intact as it fulfills the
+requirements. Bitswap usage as the backbone protocol for requesting samples and global Badger KVStore remain unaffected.
+
+### Detailed Design
+
+> All the comments on the API definitions should be preserved and potentially improved by implementations.
+
+The new block storage design is solely additive. All the existing storage-related components and functionality
+are kept with additional components introduced. Altogether, existing and new components will be recomposed to serve as the
+foundation of our improved block storage subsystem.
+
+The central data structure representing Celestia block data is EDS(`rsmt2d.ExtendedDataSquare`), and the new storage design
+is focused around storing entire EDSes as a whole rather than a set of individual chunks, s.t. storage subsystem
+can handle storing and streaming/serving blocks of 4MB and more.
+
+#### EDS Serde
```

Comment:
What does serde stand for? Shouldn't the title here be:
#### EDS (De-)Serialization